website/server/models/task.js

```diff
@@ -167,6 +182,20 @@ TaskSchema.statics.fromJSONV2 = function fromJSONV2 (taskObj) {
 
 export let Task = mongoose.model('Task', TaskSchema);
 
+Task.schema.path('shortName').validate(function valiateShortNameNotTaken (shortName, respond) {
```

Comment:
@paglias I was able to move almost all the duplicate shortName checking to the model, but because I needed access to `findOne` to do it, I had to put it after the model was defined. Is that acceptable? Or can you think of a better way to do it?